guides/07_streaming/03_object-detection-from-video.md

```diff
@@ -0,0 +1,186 @@
+            output_video = cv2.VideoWriter(output_video_name, video_codec, desired_fps, (width, height)) # type: ignore
+
+        iterating, frame = cap.read()
+        n_frames += 1
+```
+
+1. The OpenCV set-up
+
+One of the industry standards for creating videos in python is OpenCV. The syntax is a bit confusing but it's speed and popularity make it a good choice for any application.
+
+The `cap` variable is how we will read from the input video. Whenever we call `cap.read()`, we are reading the next frame in the video.
+
+In order to stream video in Gradio, we need to yield a different video file for each "chunk" of the output video.
+We create the next video file to write to with the `output_video = cv2.VideoWriter(output_video_name, video_codec, desired_fps, (width, height))` line. The `video_codec` is how we specify the type of video file. Only "mp4" and "ts" files are supported for video sreaming at the moment.
+
+
+2. The Inference Loop
+
+For each frame in the video we will resize to be half the size. OpenCV reads files in `BGR` format, so will convert to the expected `RGB` format of transfomers. That's what the first two lines of the while loop are doing. 
+
+We take every other frame and add it to a `batch` list so that the output video is half the original FPS. When the batch covers two seconds of video, we will run the model. The two second threshold was chosen to keep the processing time of each batch small enough so that video is smoothly displayed in the server while not requiring too many separate forward passes. In order for video streaming to work properly in Gradio, the batch size should be at least 1 second. 
+
+We run the forward pass of the model and then use the `post_process_object_detection` method of the model to scale the detected bounding boxes to the size of the input frame.
+
+We make use of a custom function to draw the bounding boxes (source [here](https://huggingface.co/spaces/gradio/rt-detr-object-detection/blob/main/draw_boxes.py#L14)). We then have to convert from `RGB` to `BGR` before writing back to the output video.
+
+Once we have finished processing the batch, we create a new output video file for the next batch.
+
+## The Gradio Demo
+
+The UI code is pretty similar to other kinds of Gradio apps. 
+We'll use a standard two-column layout so that users can see the input and output videos side by side.
+
+In order for streaming to work, we have to set `streaming=True` in the output video. Setting the video
+to autoplay is not necessary but it's a better experience for users.
+
+```python
+with gr.Blocks() as app:
+    gr.HTML(
+        """
+    <h1 style='text-align: center'>
+    Video Object Detection with RT-DETR
+    </h1>
+    """)
+    gr.HTML(
+        """
+        <h3 style='text-align: center'>
+        <a href='https://arxiv.org/abs/2304.08069' target='_blank'>arXiv</a> | <a href='https://huggingface.co/PekingU/rtdetr_r101vd_coco_o365' target='_blank'>github</a>
+        </h3>
+        """)
```

Comment:
replace with `gr.Markdown()` and remove the html styling for simplicity?

Reply:
I removed the second HTML component but kept the first one as I like the centered title